### PR TITLE
Test eager activity + non-remote-activities worker

### PR DESCRIPTION
**Pull request by bergundy**
Verifying that the worker doesn't crash with this combination as observed and fixed in Core based SDKs.

### CODE REVIEW

File: features/eager_activity/non_remote_activities_worker/feature.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                # Pick a long enough timeout for busy CI but not too long to get feedback quickly
+                schedule_to_close_timeout=timedelta(seconds=3),
+            )
+            raise RuntimeError("Expected activity to time out")
+        except ActivityError as e:
+            # Catch activity failure in the workflow, check that it is caused by schedule-to-start timeout
+            if not isinstance(e.cause, TimeoutError):
+                raise e
+
+            assert isinstance(e.cause, TimeoutError)
```
Comment (cretz) [inline features/eager_activity/non_remote_activities_worker/feature.py:27]: Already effectively done by the conditional above

File: features/eager_activity/non_remote_activities_worker/feature.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@activity.defn
+async def dummy() -> None:
+    pass
+
+
+# Start a worker with activities registered and non-local activities disabled
+register_feature(
+    workflows=[Workflow],
+    activities=[dummy],
+    worker_config=WorkerConfig(no_remote_activities=True),
```
Comment (cretz) [inline features/eager_activity/non_remote_activities_worker/feature.py:41]: Alternatively could have just left off `activities` here when registering, though I understand if this wants to test the core side of polling or not only

**Review by cretz — APPROVED** [review]

File: features/eager_activity/non_remote_activities_worker/feature.py (hunk 2 above)
Comment (bergundy) [reply features/eager_activity/non_remote_activities_worker/feature.py:41]: Yeah, this tests the case where the worker supports local activities and no remote activities.

File: features/eager_activity/non_remote_activities_worker/feature.py (hunk 1 above)
Comment (bergundy) [reply features/eager_activity/non_remote_activities_worker/feature.py:27]: I needed this one for mypy to infer the type, can remove the condition above.

**Comment by bergundy** [comment]
Merging even though some of the tests are flaky.
The flakes are not related to my changes.

I'll open an issue to fix the flakes.

**Comment by Spikhalskiy** [comment]
This PR has a stable failing test for JavaSDK: eager_activity/non_remote_activities_worker

**Pull request merged**